network-store-iidm-impl/src/test/java/com/powsybl/network/store/iidm/impl/OperatingStatusExtensionTest.java

```diff
@@ -163,4 +166,104 @@ public void testHvdcLineOperatingStatusExtension() {
+        network.addListener(listener1);
+        network.addListener(listener2);
+
+        Line l1 = network.getLine("L1");
+        l1.newExtension(OperatingStatusAdder.class)
+                .withStatus(OperatingStatus.Status.PLANNED_OUTAGE)
+                .add();
+
+        assertEquals(OperatingStatus.Status.PLANNED_OUTAGE, l1.getExtension(OperatingStatus.class).getStatus());
+
+        assertEquals(0, listener1.getNbRemovedExtension());
+        l1.removeExtension(Measurements.class);
+        assertEquals(0, listener1.getNbRemovedExtension());
+        l1.removeExtension(OperatingStatus.class);
+        assertEquals(1, listener1.getNbRemovedExtension());
+
+        assertNull(l1.getExtension(OperatingStatus.class));
+    }
+
+    private class DummyNetworkListener implements NetworkListener {
+
+        private int nbRemovedExtension = 0;
+
+        public int getNbRemovedExtension() {
+            return nbRemovedExtension;
+        }
+
+        @Override
+        public void onExtensionAfterRemoval(Identifiable<?> identifiable, String extensionName) {
+            nbRemovedExtension++;
+        }
+
+        @Override
+        public void onExtensionBeforeRemoval(Extension<?> extension) {
+        }
+
+        @Override
+        public void onCreation(Identifiable identifiable) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void beforeRemoval(Identifiable identifiable) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void afterRemoval(String id) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void onUpdate(Identifiable identifiable, String s, Object o, Object o1) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void onVariantCreated(String sourceVariantId, String targetVariantId) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void onVariantRemoved(String variantId) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void onUpdate(Identifiable<?> identifiable, String attribute, String variantId, Object oldValue,
+                             Object newValue) {
+            throw new UnsupportedOperationException("Unimplemented method 'onUpdate'");
+        }
+
+        @Override
+        public void onExtensionCreation(Extension<?> extension) {
+            throw new UnsupportedOperationException("Unimplemented method 'onExtensionCreation'");
+        }
+
+        @Override
+        public void onExtensionUpdate(Extension<?> extendable, String attribute, Object oldValue, Object newValue) {
```

Comment:
```suggestion
        public void onExtensionUpdate(Extension<?> extension, String attribute, Object oldValue, Object newValue) {
```
Like in the previous function, it makes more sense to me.

Reply:
done